shuffle/cli/tests/shuffle-forge-cli-tests.rs

```diff
@@ -0,0 +1,282 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use forge::{
+    forge_main, AdminContext, AdminTest, ForgeConfig, LocalFactory, Options, Result, Test,
+};
+
+use shuffle::forge_test_helper::{bootstrap_shuffle, ShuffleTestHelper};
+use smoke_test::scripts_and_modules::enable_open_publishing;
+use std::{
+    os::unix::prelude::ExitStatusExt,
+    process::{ExitStatus, Stdio},
+    thread, time,
+};
+use tempfile::tempdir;
+use tokio::runtime::Runtime;
+
+const HOME_BASE_PATH: &str = "--home-base-path";
+pub const BINARY: &str = env!("CARGO_BIN_EXE_shuffle");
+
+fn main() -> Result<()> {
```

Comment:
While it may seem great to have everything in `shuffle/cli/tests/` we now have multiple entry points in this crate:
1. Regular expected rust tests
2. shuffle-forge-cli-tests
3. shuffle-forge-integration-tests

My concern is that we're conflating a lot of tests into one folder, but there is a benefit to running one `cargo test` for all of this.

Reply:
An example of where this bites us is in CI. forge tests are run in different jobs/tasks than unit tests

<img width="318" alt="CleanShot 2021-11-22 at 08 56 15@2x" src="https://user-images.githubusercontent.com/635121/142902853-9cfd637d-e600-49b5-9acf-3a45a6c7924a.png">

How do we resolve this CI issue? Has that been investigated?

https://github.com/dimroc/diem/blob/c1416df8510c7e2d93ccf60f5d33341c5b7db199/x.toml#L4-L8
```toml
[system-tests]
jsonrpc-integration-tests = { path = "json-rpc/integration-tests" }
shuffle-integration-tests = { path = "shuffle/integration-tests" }
smoke-test = { path = "smoke-test" }
testcases = { path = "testsuite/testcases" }
```